nixos/modules/system/boot/systemd.nix

```diff
@@ -888,9 +890,13 @@ in
       let
         type = service.serviceConfig.Type or "";
         restart = service.serviceConfig.Restart or "no";
-      in optional
+        hasDeprecated = builtins.hasAttr "StartLimitInterval" service.serviceConfig;
+      in (optional
       (type == "oneshot" && (restart == "always" || restart == "on-success"))
       "Service '${name}.service' with 'Type=oneshot' cannot have 'Restart=always' or 'Restart=on-success'")
+      ++ (optional hasDeprecated ''Service '${name}.service' uses the attribute
+      StartLimitInterval in the Service section, which is deprecated. See
+      https://github.com/NixOS/nixpkgs/issues/45786.''))
```

Comment:
I don't know how warnings are rendered, but doesn't using a `''` as done here insert spaces `      ` in the middle of the message ([logic from the manual](https://nixos.org/manual/nix/stable/#idm140737322059824))?

E.g. should this be:

```
... hasDeprecated ''
  Service '${name}.service' uses the attribute
  StartLimitInterval in the Service section, which is deprecated. See
  https://github.com/NixOS/nixpkgs/issues/45786.
''
```

Reply:
I tried it, indeed the formatting is odd this way:

```
trace: warning: Service 'display-manager.service' uses the attribute
      StartLimitInterval in the Service section, which is deprecated. See
      https://github.com/NixOS/nixpkgs/issues/45786.
```

Reply:
Suggestion for the whole `warnings` block:

```nix
    warnings = concatLists (
      mapAttrsToList
        (name: service:
          let
            type = service.serviceConfig.Type or "";
            restart = service.serviceConfig.Restart or "no";
            hasDeprecated = builtins.hasAttr "StartLimitInterval" service.serviceConfig;
          in
            concatLists [
              (optional (type == "oneshot" && (restart == "always" || restart == "on-success"))
                "Service '${name}.service' with 'Type=oneshot' cannot have 'Restart=always' or 'Restart=on-success'"
              )
              (optional hasDeprecated
                "Service '${name}.service' uses the attribute 'StartLimitInterval' in the Service section, which is deprecated. See https://github.com/NixOS/nixpkgs/issues/45786."
              )
            ]
        )
        cfg.services
    );
```

Reply:
I'll force push a refactor that does this in a moment.

Reply:
OK done. Now it renders correctly and the diff looks much nicer.